pkg/k8shandler/exec.go

```diff
@@ -0,0 +1,81 @@
+package k8shandler
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sclient"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+type ExecConfig struct {
+	namespace      string
+	labelName      string
+	labelPrefix    string
+	containerName  string
+	command        []string
+	kubeConfigPath string
+	masterURL      string
+}
+
+func PodExec(config *ExecConfig) (*bytes.Buffer, *bytes.Buffer, error) {
+	var (
+		execOut bytes.Buffer
+		execErr bytes.Buffer
+		esPod   *v1.Pod
+	)
+
+	client := k8sclient.GetKubeClient()
+
+	pods, err := client.CoreV1().Pods(config.namespace).List(metav1.ListOptions{})
```

Comment:
Can we not get the list of ES pods in the initial request by parameters provided to the list call so you don't have to filter in subsequent code?

Reply:
We know the pod name.
Maybe that will be enough..

Reply:
Isn't this code getting the list of all pods to try and find an eligible node from which to get one ES node?  We don't need a specific name, but if you do already know the name it would be faster to use it.